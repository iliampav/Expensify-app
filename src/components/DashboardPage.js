import React from 'react';

const ExpenseDashboardPage = () => (
    <div>
        <p> Dashboard Page Content</p>
    </div>
)

export default ExpenseDashboardPage;