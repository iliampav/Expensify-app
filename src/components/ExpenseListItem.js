import React from 'react';
import { Link } from 'react-router-dom';
 
const ExpenseListItem = ({dispatch, id, description, amount, createdAt}) => (
    //lista dos itens
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>{amount} - {createdAt}</p>    
    </div>
)

export default ExpenseListItem