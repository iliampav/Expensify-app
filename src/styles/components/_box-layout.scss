.box-layout {
    background-image: url("../../images/bg.jpg");
}