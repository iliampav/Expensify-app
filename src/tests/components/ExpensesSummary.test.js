import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should correctly render expenses Sumary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={325}/>)
    expect(wrapper).toMatchSnapshot();
})

test('Should correctly render expenses Sumary with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={3254577542}/>)
    expect(wrapper).toMatchSnapshot();
})