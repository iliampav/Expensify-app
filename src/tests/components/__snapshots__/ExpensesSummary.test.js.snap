// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expenses Sumary with multiple expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content_container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       
      expenses
       totalling 
      <span>
        32,545,775.42
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should correctly render expenses Sumary with one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content_container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        3.25
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
